### PR TITLE
added verifier check in sticker, fixed the edge case in product

### PR DESCRIPTION
**Pull request by Soleimani193**
This PR adds the verifier check for the local opening in the stitcker compiler.
It also addresses the edge case 0^0=1 in the symbolic.NewProduct()

### Checklist

* [ ] I wrote new tests for my new core changes.
* [ ] I have successfully ran tests, style checker and build against my new changes locally.
* [ ] I have informed the team of any breaking changes if there are any.

### CODE REVIEW

**Review by gusiri — APPROVED** [review]
LGTM

File: prover/protocol/compiler/splitter/sticker/sticker.go [hunk 1]
```diff
@@ -434,6 +436,22 @@ func (ctx *stickContext) compileFixedEvaluation() {
...
 			run.AssignLocalPoint(newQ.ID, y)
 		})
+
+		// The verifier ensures that the old and new queries have the same assignement
+		ctx.comp.InsertVerifier(round, func(run *wizard.VerifierRuntime) error {
+			oldParams := run.GetLocalPointEvalParams(q.ID)
+			newParams := run.GetLocalPointEvalParams(queryName(q.ID))
+
+			if oldParams != newParams {
+				return fmt.Errorf("sticker verifier failed for local opening %v - %v", q.ID, queryName(q.ID))
```
Comment (gusiri) [inline prover/protocol/compiler/splitter/sticker/sticker.go:446]: I think we use `logrus.Errorf` instead of `fmt.Errof`

**Review by AlexandreBelling — APPROVED** [review]
